Description: Un needed linking with lX11
Author: Scott Howard <showard@debian.org>
Index: bossa/Makefile
===================================================================
--- bossa.orig/Makefile	2014-02-01 12:47:07.215481575 -0500
+++ bossa/Makefile	2014-02-01 12:47:07.207481575 -0500
@@ -66,7 +66,6 @@
 ifeq ($(OS),Linux)
 COMMON_SRCS+=PosixSerialPort.cpp LinuxPortFactory.cpp
 COMMON_LIBS=-Wl,--as-needed
-WX_LIBS+=-lX11
 
 MACHINE:=$(shell uname -m)
 
